src/main/java/StringAddCalculator.java

```diff
@@ -0,0 +1,56 @@
+import java.util.Arrays;
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class StringAddCalculator {
+
+    private static final String DELIMITER = "[,:;]";
+    private static final String DELIMITER_REGEX = "//(.)\n(.*)";
+    private static final Pattern PATTERN = Pattern.compile(DELIMITER_REGEX);
+
+    public static int splitAndSum(String text) {
+        if (Objects.isNull(text) || text.isBlank()) {
+            return 0;
+        }
+        return sum(toInt(split(text)));
+    }
+
+    private static String[] split(String text) {
+        Matcher matcher = PATTERN.matcher(text);
+        if (!matcher.find()) {
+            return text.split(DELIMITER);
+        }
+
+        String customDelimiter = matcher.group(1);
+        String delimitedText = matcher.group(2);
+        return delimitedText.split(customDelimiter);
+    }
+
+    private static int[] toInt(String[] texts) {
+        return Arrays.stream(texts)
+            .mapToInt(StringAddCalculator::toIntOrThrow)
+            .toArray();
+    }
+
+    private static int toIntOrThrow(String text) {
+        try {
+            int parsedValue = toInt(text);
+            if (parsedValue < 0) {
+                throw new NumberFormatException();
+            }
```

Comment:
`NumberFormatException` 은 숫자가 아닌 경우일 때 사용하는 예외입니다 :)
음수는 숫자이기 때문에 올바른 예외는 아닐것 같아요! 🤔

`음수를 전달할 경우 RuntimeException 예외가 발생해야 한다. (예 : “-1,2,3”)` 이 요구사항을 만족하기 위해 바로 예외를 전파하는 방법은 어떨까요?

```suggestion
            if (parsedValue < 0) {
                throw new RuntimeException();
            }
```

Reply:
넵 해당 조건에 적절한 예외인지에 대한 판단을 놓친 것 같습니다. 말씀해주신대로 수정하겠습니다. 감사합니다!